internal/service/base/resource_language_translation.go

```diff
@@ -0,0 +1,573 @@
+						},
+						"reference_text": schema.StringAttribute{
+							Computed:    true,
+							Description: "The English string text associated with the UI element.",
+						},
+						"short_key": schema.StringAttribute{
+							Computed:    true,
+							Description: "The interface element only (for example, `button.cancel`).",
+						},
+						"translated_text": schema.StringAttribute{
+							Computed:    true,
+							Optional:    true,
+							Description: "The translated string text associated with the UI element. Excluding this attribute will reset the translation to its original configuration.",
+						},
+					},
+				},
+				Required: true,
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+		},
+	}
+}
+
+func (model *languageTranslationsResourceModel) buildClientStruct() (*[]management.LocaleTranslation, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	localeTranslationSlice := &[]management.LocaleTranslation{}
+
+	if !model.Translations.IsNull() || !model.Translations.IsUnknown() {
+		for _, translation := range model.Translations.Elements() {
+			innerLocaleTranslation := management.LocaleTranslation{}
+			innerTranslationAttrs := translation.(types.Object).Attributes()
+
+			if innerTranslationAttrs["translated_text"] != nil &&
+				!innerTranslationAttrs["translated_text"].(types.String).IsNull() &&
+				!innerTranslationAttrs["translated_text"].(types.String).IsUnknown() {
+				innerLocaleTranslation.TranslatedText = innerTranslationAttrs["translated_text"].(types.String).ValueString()
+			}
+			innerLocaleTranslation.Key = innerTranslationAttrs["key"].(types.String).ValueString()
+			innerLocaleTranslation.ReferenceText = innerTranslationAttrs["reference_text"].(types.String).ValueStringPointer()
+			innerLocaleTranslation.ShortKey = innerTranslationAttrs["short_key"].(types.String).ValueStringPointer()
+
+			*localeTranslationSlice = append(*localeTranslationSlice, innerLocaleTranslation)
+		}
+	}
+
+	return localeTranslationSlice, diags
+}
+
+// Build a default client struct to reset the resource to its default state
+func (model *languageTranslationResource) buildDefaultClientStruct(p languageTranslationsResourceModel) *[]management.LocaleTranslation {
+	// Sending only the key will reset the translation to its original configuration - per API documentation
+	result := &[]management.LocaleTranslation{}
+	for _, translation := range p.Translations.Elements() {
+		innerTranslationAttrs := translation.(types.Object).Attributes()
+		innerLocaleTranslation := management.LocaleTranslation{
+			Key: innerTranslationAttrs["key"].(types.String).ValueString(),
+		}
+
+		*result = append(*result, innerLocaleTranslation)
+	}
+
+	return result
+}
+
+func (state *languageTranslationsResourceModel) readClientResponse(response []management.LocaleTranslation, data languageTranslationsResourceModel) diag.Diagnostics {
```

Comment:
You can use `state` directly in this function, you don't need to pass in the `data` parameter.